mitmproxy/tools/console/grideditor/editors.py

```diff
@@ -175,11 +175,23 @@ def is_error(self, col, val):
 class DataViewer(base.GridEditor, layoutwidget.LayoutWidget):
     title = None  # type: str
 
-    def __init__(self, master, vals):
+    def __init__(
+            self,
+            master,
+            vals: typing.Union[
+                typing.List[typing.List[typing.AnyStr]],
+                typing.List[typing.AnyStr],
+                str,
+            ]):
         if vals:
+            # Whatever vals is, make it a list of rows containing lists of column values.
+            if isinstance(vals, str):
+                vals = [vals]
```

Comment:
This is an additional bugfix, where we previously rendered a single string as one character per row.